sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkProcessingUtil.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.batch.ItemBulkOperation;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.Beta;
+import reactor.core.publisher.Flux;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Utility for creating bulk operations which can be executed by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)} .
+ *
+ * Also while creating these operation, if some options which are only for individual operation can be provided by passing
+ * a {@link BulkItemRequestOptions} while creating the bulk operation.
+ *
+ * See also {@link BulkProcessingOptions}.
+ */
+@Beta(Beta.SinceVersion.V4_7_0)
+public final class BulkProcessingUtil {
```

Comment:
@FabianMeiswinkel @kushagraThapar @kirankumarkolli thought on the naming?

Reply:
My preference would be:

BulkOperations as the class name

Factory method names as newCreateItemOperation(T item, PartitionKey partitionKey) or newReadItemOperation(String id, PartitionKey partitionKey)

I think the *bulk* can get removed because that is clear from the class name of the factory

I like the newXXX naming convention for factory methods beter than createXXX - but would prefer either of them over getXXX

Reply:
Have changed class to BulkOperations, seems better actually. Have changed function names too. CreateXXX is confusing to me too as we have create as an operation too. newXXX seems fine, so changed it for now.

Reply:
LGTM now - thanks!

Reply:
`BulkOperations` looks good to me too.